face/infra/roc-v3/src/main/java/com/simprints/face/infra/rocv3/matching/RocV3Matcher.kt

```diff
@@ -1,46 +1,53 @@
 import ai.roc.rocsdk.embedded.roc
-import ai.roc.rocsdk.embedded.rocConstants.ROC_FACE_FAST_FV_SIZE
+import ai.roc.rocsdk.embedded.rocConstants
 import com.simprints.core.ExcludedFromGeneratedTestCoverageReports
+import com.simprints.face.infra.basebiosdk.matching.FaceIdentity
 import com.simprints.face.infra.basebiosdk.matching.FaceMatcher
-import com.simprints.face.infra.rocv3.detection.RocV3Detector.Companion.RANK_ONE_TEMPLATE_FORMAT_3_1
-import javax.inject.Inject
-import javax.inject.Singleton
-
-@Singleton
-class RocV3Matcher @Inject constructor() : FaceMatcher() {
-    override val matcherName
-        get() = "RANK_ONE"
-
-    override val supportedTemplateFormat
-        get() = RANK_ONE_TEMPLATE_FORMAT_3_1
-
-    // Ignore this method from test coverage calculations
-    // because it uses jni native code which is hard to test
-    @ExcludedFromGeneratedTestCoverageReports(
-        reason = "This function uses roc class that has native functions and can't be mocked",
-    )
-    override suspend fun getComparisonScore(
-        probe: ByteArray,
-        matchAgainst: ByteArray,
-    ): Float {
-        val probeTemplate = roc.new_uint8_t_array(ROC_FACE_FAST_FV_SIZE.toInt())
-        roc.memmove(roc.roc_cast(probeTemplate), probe)
+import com.simprints.face.infra.basebiosdk.matching.FaceSample
+
+@ExcludedFromGeneratedTestCoverageReports(
+    reason = "This function uses roc class that has native functions and can't be mocked",
+)
+class RocV3Matcher(
+    override val probeSamples: List<FaceSample>
+) : FaceMatcher(probeSamples) {
+
+    var probeTemplates: List<SWIGTYPE_p_unsigned_char> = probeSamples.mapIndexed { i, probe ->
+        val probeTemplate: SWIGTYPE_p_unsigned_char =
+            roc.new_uint8_t_array(rocConstants.ROC_FACE_FAST_FV_SIZE.toInt())
+        roc.memmove(roc.roc_cast(probeTemplate), probe.template)
+        probeTemplate
+    }
 
-        val matchTemplate = roc.new_uint8_t_array(ROC_FACE_FAST_FV_SIZE.toInt())
-        roc.memmove(roc.roc_cast(matchTemplate), matchAgainst)
+    override suspend fun getHighestComparisonScoreForCandidate(candidate: FaceIdentity): Float =
+        probeTemplates.flatMap { probeTemplate ->
+            candidate.faces.map { face ->
+                getSimilarityScoreForCandidate(probeTemplate, face.template)
+            }
+        }.max()
+
+    private fun getSimilarityScoreForCandidate(
+        probeTemplate: SWIGTYPE_p_unsigned_char,
+        candidateTemplate: ByteArray,
+    ): Float {
+        val matchTemplate = roc.new_uint8_t_array(rocConstants.ROC_FACE_FAST_FV_SIZE.toInt())
+        roc.memmove(roc.roc_cast(matchTemplate), candidateTemplate)
 
         val similarity = roc.roc_embedded_compare_templates(
             probeTemplate,
-            ROC_FACE_FAST_FV_SIZE,
+            rocConstants.ROC_FACE_FAST_FV_SIZE,
             matchTemplate,
-            ROC_FACE_FAST_FV_SIZE,
+            rocConstants.ROC_FACE_FAST_FV_SIZE,
```

Comment:
Can we not add rocConstants 😇

Reply:
It is just a matter of imports since those are defined in the SDK, I can make more explicit import.